theme/src/components/wrap-root-element.js

```diff
@@ -32,6 +33,9 @@ const components = {
   ul: List,
   ol: List.withComponent('ol'),
   dl: DescriptionList,
+
+  // Shortcodes (https://mdxjs.com/blog/shortcodes)
+  Note,
```

Comment:
Making `Note` a [shortcode](https://mdxjs.com/blog/shortcodes) allows people to use it in their markdown files without having to import it. I think this is a huge UX improvement. We should probably make `Caption` and `Do/Dont` shortcodes as well. Although, I intentionally didn't make those other components shortcodes in this PR to keep these changes focused on the `Note` component.